ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/constraint/ConstraintMapperInterface.java

```diff
@@ -0,0 +1,23 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.openapi.service.mapper.constraint;
+
+public interface ConstraintMapperInterface {
```

Comment:
Since there are several use interfaces with `MapperInferface` suffice with only one method can't we create a generic interface like `OASMapper` and then implement it for the relevant use cases.
```java
public interface OASMapper {
  void addMapping();
}

public class ServerMapper implements OASMapper {}

public class ConstraintMapper implements OASMapper {}
```

WDYT ?

Reply:
The future plan is to move all of these mappers to separate Java packages. So IMO this is not necessary

Reply:
But, now we are creating new interface for each functionality. Rather than that I feel this would be much cleaner.

Reply:
Yeah, having a common interface much cleaner. but for the long run, we have a plan to extend support for various OAS versions. At that time, I think this current implementation (separate interface) will be scalable across different versions, as each component has a unique implementation, and some of the logic may become redundant. WDYT?

Reply:
Agree, will keep it as it is for now